core/src/main/java/ar/com/wolox/wolmo/core/util/ImageProvider.kt

```diff
@@ -0,0 +1,247 @@
+            }
+
+            return Bitmap.createScaledBitmap(image, finalWidth, finalHeight, true)
+        }
+
+        /**
+         * Get [byte[]] from an image [File]
+         *
+         * @param file      target image file
+         * @param format    image compress format
+         * @param quality   compress quality, between 0 and 100
+         * @param maxWidth  max width of the target image
+         * @param maxHeight max height of the target image
+         * @return byte array with the formatted information of the image file, if the image exceeded
+         * boundaries, it's re scaled.
+         */
+        fun getImageAsByteArray(
+                file: File,
+                format: Bitmap.CompressFormat,
+                @IntRange(from = 0, to = 100) quality: Int,
+                maxWidth: Int,
+                maxHeight: Int): ByteArray {
+
+            return getImageAsByteArray(
+                    BitmapFactory.decodeFile(file.path),
+                    format,
+                    quality,
+                    maxWidth,
+                    maxHeight)
+        }
+
+        /**
+         * Get [byte[]] representation from a [Bitmap], with boundaries.
+         *
+         * @param bitmap    target Bitmap
+         * @param format    image compress format
+         * @param quality   compress quality, between 0 and 100
+         * @param maxHeight max height of the target image
+         * @param maxWidth  max width of the target image
+         * @return byte array with the formatted information of the Bitmap, if the image exceeded
+         * boundaries, it's re scaled.
+         */
+        fun getImageAsByteArray(
+                bitmap: Bitmap,
+                format: Bitmap.CompressFormat,
+                @IntRange(from = 0, to = 100) quality: Int,
+                maxWidth: Int,
+                maxHeight: Int): ByteArray {
+
+            val targetBitmap = fit(bitmap, maxWidth, maxHeight)
+
+            val bytes = ByteArrayOutputStream()
+            targetBitmap.compress(format, sanitizeQuality(quality), bytes)
+
+            return bytes.toByteArray()
+        }
+    }
+
+    /**
+     * Triggers an intent to go to the device's image gallery and returns an URI with the file.
+     *
+     *
+     *
+     *
+     * Override the onActivityResult method in your fragment and specify behaviour
+     * for the provided request code. The selected image URI will be
+     * returned in the data variable of the activity result.
+     *
+     * @param fragment    A fragment where the get image intent is going to be called
+     * @param requestCode of the result call to be identified with
+     * @param errorResId  [StringRes] to be displayed in case of error
+     */
+    fun getImageFromGallery(fragment: Fragment, requestCode: Int, @StringRes errorResId: Int) {
+        val i = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI)
```

Comment:
I think since we are refactoring all, we could use more expressive names 👀 .

Reply:
ok :+1: